alvr/server_core/src/tracking/mod.rs

```diff
@@ -0,0 +1,495 @@
+            last_face_data: FaceData::default(),
+        }
+    }
+
+    pub fn recenter(
+        &mut self,
+        position_recentering_mode: PositionRecenteringMode,
+        rotation_recentering_mode: RotationRecenteringMode,
+    ) {
+        let position = match position_recentering_mode {
+            PositionRecenteringMode::Disabled => Vec3::ZERO,
+            PositionRecenteringMode::LocalFloor => {
+                let mut pos = self.last_head_pose.position;
+                pos.y = 0.0;
+
+                pos
+            }
+            PositionRecenteringMode::Local { view_height } => {
+                self.last_head_pose.position
+                    - self.last_head_pose.orientation * Vec3::new(0.0, view_height, 0.0)
+            }
+        };
+
+        let orientation = match rotation_recentering_mode {
+            RotationRecenteringMode::Disabled => Quat::IDENTITY,
+            RotationRecenteringMode::Yaw => {
+                let mut rot = self.last_head_pose.orientation;
+                // extract yaw rotation
+                rot.x = 0.0;
+                rot.z = 0.0;
+                rot = rot.normalize();
+
+                rot
+            }
+            RotationRecenteringMode::Tilted => self.last_head_pose.orientation,
+        };
+
+        self.inverse_recentering_origin = Pose {
+            position,
+            orientation,
+        }
+        .inverse();
+    }
+
+    pub fn recenter_pose(&self, pose: Pose) -> Pose {
+        self.inverse_recentering_origin * pose
+    }
+
+    pub fn recenter_motion(&self, motion: DeviceMotion) -> DeviceMotion {
+        DeviceMotion {
+            pose: self.recenter_pose(motion.pose),
+            linear_velocity: self.inverse_recentering_origin.orientation * motion.linear_velocity,
+            angular_velocity: self.inverse_recentering_origin.orientation * motion.angular_velocity,
+        }
+    }
+
+    // Performs all kinds of tracking transformations, driven by settings.
+    pub fn report_device_motions(
+        &mut self,
+        config: &HeadsetConfig,
+        timestamp: Duration,
+        device_motions: &[(u64, DeviceMotion)],
+    ) {
+        let mut device_motion_configs = HashMap::new();
+        device_motion_configs.insert(*HEAD_ID, MotionConfig::default());
+        device_motion_configs.extend([
+            (*BODY_CHEST_ID, MotionConfig::default()),
+            (*BODY_HIPS_ID, MotionConfig::default()),
+            (*BODY_LEFT_ELBOW_ID, MotionConfig::default()),
+            (*BODY_RIGHT_ELBOW_ID, MotionConfig::default()),
+            (*BODY_LEFT_KNEE_ID, MotionConfig::default()),
+            (*BODY_LEFT_FOOT_ID, MotionConfig::default()),
+            (*BODY_RIGHT_KNEE_ID, MotionConfig::default()),
+            (*BODY_RIGHT_FOOT_ID, MotionConfig::default()),
+        ]);
+
+        if let Switch::Enabled(controllers) = &config.controllers {
+            let t = controllers.left_controller_position_offset;
```

Comment:
why t and not p

Reply:
Grubhub copilot

Reply:
I got my AI off of Wish

Reply:
Nevermind, this was my code, and also quite old. I used T as for translation

Reply:
That's what I assumed, but it makes little sense to pipe that into a position field without any actual translating being done there. But your call whether to fix it or just leave it be

Reply:
I'd leave it be tbh